src/coll_select/coll_select.c

```diff
@@ -0,0 +1,661 @@
+            d = &new->list_elem;
+            s = s->next;
+        }
+    }
+    /* Merge consequtive ranges with the same score and same init fn
+       if any have been produced by the algorithm above */
+    ucc_msg_range_t *range, *tmp, *next;
+    ucc_list_for_each_safe(range, tmp, dest, list_elem) {
+        if (range->list_elem.next != dest) {
+            next = ucc_container_of(range->list_elem.next, ucc_msg_range_t,
+                                    list_elem);
+            if (range->score == next->score && range->end == next->start &&
+                range->init == next->init && range->team == next->team) {
+                next->start = range->start;
+                ucc_list_del(&range->list_elem);
+                ucc_free(range);
+            }
+        }
+    }
+    /* remove potentially disabled ranges */
+    ucc_list_for_each_safe(range, tmp, dest, list_elem) {
+        if (0 == range->score) {
+            ucc_list_del(&range->list_elem);
+            ucc_free(range);
+        }
+    }
+    return UCC_OK;
+out:
+    return status;
+}
+
+ucc_status_t ucc_coll_score_update(ucc_coll_score_t *score,
+                                   ucc_coll_score_t *update)
+{
+    ucc_status_t      status;
+    int               i, j;
+    for (i = 0; i < UCC_COLL_TYPE_NUM; i++) {
+        for (j = 0; j < UCC_MEMORY_TYPE_LAST; j++) {
+            status = ucc_coll_score_update_one(&score->scores[i][j],
+                                               &update->scores[i][j]);
+            if (UCC_OK != status) {
+                return status;
+            }
+        }
+    }
+    return UCC_OK;
+}
+
+ucc_status_t ucc_coll_score_update_from_str(const char *str,
+                                            ucc_coll_score_t *score,
+                                            ucc_rank_t        team_size)
+{
+    ucc_status_t      status;
+    ucc_coll_score_t *score_str;
+    status = ucc_coll_score_alloc_from_str(str, &score_str, team_size);
+    if (UCC_OK != status) {
+        return status;
+    }
+    status = ucc_coll_score_update(score, score_str);
+    ucc_coll_score_free(score_str);
+    return status;
+}
+
+ucc_status_t ucc_coll_score_build_default(ucc_base_team_t        *team,
+                                          ucc_score_t             default_score,
+                                          ucc_base_coll_init_fn_t default_init,
+                                          ucc_coll_type_t         coll_types,
+                                          ucc_memory_type_t      *mem_types,
+                                          int mt_n, ucc_coll_score_t **score_p)
+{
+    ucc_coll_score_t *score;
+    ucc_status_t      status;
+    ucc_memory_type_t m;
+    uint64_t          c;
+    status = ucc_coll_score_alloc(&score);
+    if (UCC_OK != status) {
+        return status;
+    }
+
+    if (!mem_types)
+        mt_n = UCC_MEMORY_TYPE_LAST;
+    ucc_for_each_bit(c, coll_types) {
```

Comment:
alignment?

Reply:
no, this is a next statement after "if". Added {} to the if - now it looks clean.

Reply:
can you add { } for above if

Reply:
i did